lib/gnat.ex

```diff
@@ -37,11 +37,16 @@ defmodule Gnat do
     end
   end
 
+  defp process_message({:msg, topic, sid, body}, state) do
```

Comment:
I definitely like this function with pattern matching more than a `case` statement used in your other PR